recirq/qml_lfe/learn_dynamics_q.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Draw two copy data from 1D random scrambling or tsym style circuits.
+
+learn_dynamics_q.py --n=6 --depth=5 --n_data=20 --batch_size=5 --n_shots=1000
+
+Will create 40 total circuits, 20 scrambling and 20 tsym circuits all on
+6 qubits with depth 5. Once the circuits are generated, batch_size circuits
+will be sent for simulation/execution, drawing n_shots samples from each
+one using `run_batch` in Cirq. By default the bitstring data will be saved
+in the data folder. One can also set `use_engine` to True in order to
+run this against a processor on quantum engine.
+"""
+
+from typing import List
+
+import os
+import cirq
+import cirq_google
+import numpy as np
+from . import circuit_blocks
+
+from absl import app
+from absl import flags
+from absl import logging
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer("n", None, "System size (total qubits == 2 * n).")
+flags.DEFINE_integer("depth", None, "Circuit depth (block-wise).")
+flags.DEFINE_integer(
+    "n_data",
+    20,
+    "Number of circuits generated for each class (total circuits == 2 * n_data).",
+)
+
+flags.DEFINE_integer(
+    "batch_size",
+    5,
+    "Number of circuits to send over the wire per batch (value does not affect results).",
+)
+
+flags.DEFINE_integer(
+    "n_shots", 2000, "Number of measurements to draw from each individual circuit."
+)
+
+flags.DEFINE_string(
+    "save_dir", "./data", "Path to save experiment data (must already exist)."
+)
+
+flags.DEFINE_bool("use_engine", False, "Whether or not to use quantum engine.")
+
+
+def _flatten_circuit(circuit: cirq.Circuit) -> cirq.Circuit:
```

Comment:
Optional: Should we split this out into a common file?

Reply:
Split out.